z2/src/converter.rs

```diff
@@ -355,11 +344,45 @@ pub async fn convert_persistence(
                     }
                 };
 
+                transactions_trie
```

Comment:
Maybe insert a comment describing the process here - you look to be building your hashes in order so that you break the circular hashing dependency by assuming all hashes are 0, then performing the circular hash ops in a defined order. Leave a comment saying that's what you're doing and which order you think you are doing it in (or I can't check if you are!)

Reply:
How about I file a followup issue and we can merge this PR and move on with our lives and I'll come back later and write a comment?

Reply:
Cool :-) - https://github.com/Zilliqa/zq2/issues/1088